ldap/client_exported_test.go

```diff
@@ -0,0 +1,124 @@
+package ldap_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/cap/ldap"
+	"github.com/hashicorp/go-hclog"
+	"github.com/jimlambrt/gldap"
```

Comment:
I might suggest glap could be useful just in the vault unit tests for the ldap secrets backend and the built in ldap auth method, since it would allow those tests to be a bit more exhaustive.   with that said, I'm hopeful we can bring incorporate all of vault's needs into this package and subsequently refactor the vault bits to use this pkg.

Reply:
also, I started gldap with a simple goal of making a test directory like the OIDC testing provider possible and something like 10k LOC and 60+ commits later... 😂

Reply:
Ya, I was also thinking this could be useful for testing in Vault. I'll keep that in mind when we look closer at adopting this 👍